rust/agama-lib/src/software/client.rs

```diff
@@ -32,12 +32,18 @@ pub enum SelectedBy {
     None = 2,
 }
 
-impl From<u8> for SelectedBy {
-    fn from(value: u8) -> Self {
+#[derive(Debug, thiserror::Error)]
+#[error("Unknown selected by value: '{0}'")]
+pub struct UnknownSelectedBy(u8);
+
+impl TryFrom<u8> for SelectedBy {
+    type Error = UnknownSelectedBy;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
         match value {
-            0 => Self::User,
-            1 => Self::Auto,
-            _ => Self::None,
+            0 => Ok(Self::User),
+            1 => Ok(Self::Auto),
```

Comment:
now, I am confused, why is not there `2 => Ok(Self::None)`

Reply:
Because it is not expected to get a `2` from the API (it does not exist).

Reply:
ah, ok.